tests/testDCSAM.cpp

```diff
@@ -1344,6 +1344,212 @@ TEST(TestSuite, dcMaxMixture_semantic_slam) {
   EXPECT_EQ(mpeClassL1, 1);
 }
 
+/**
+ * This is for testing the behavior of factor removal
+ */
+TEST(TestSuite, factor_removal) {
```

Comment:
I think we can prefer for this to be a simpler test with a known solution that we can check with `EXPECT_EQ` rather than a qualitative one. We can remove all the plotting and most of the graph construction code and just test that the main factor removal feature is functioning as anticipated.